tests/diff.js

```diff
@@ -67,4 +67,88 @@ describe( 'diff', () => {
 		diff( getLongText( 10 ), getLongText( 1990 ) );
 		testUtils.sinon.assert.called( fastDiffSpy );
 	} );
+
+	describe( 'with multi-byte unicode', () => {
+		describe( 'simple emoji - single unicode code point', () => {
+			// 🙂 = '\ud83d\ude42' = 2 chars
+			const emojiLength = '🙂'.length;
+			const emojiDiffInsert = new Array( emojiLength ).fill( 'insert' );
+			const emojiDiffEqual = new Array( emojiLength ).fill( 'equal' );
+			const emojiDiffDelete = new Array( emojiLength ).fill( 'delete' );
+
+			it( 'should properly handle emoji insertion', () => {
+				expect( diff( 'abc', 'ab🙂c' ) ).to.deep.equals( [ 'equal', 'equal', ...emojiDiffInsert, 'equal' ] );
```

Comment:
Frankly speaking, this `emojiDiffInsert` makes these tests much harder to read than they need to be. IMO, assertions should be as "raw" as possible so you don't need to jump to e.g. a definition of a variable to understand them. Here, I needed to scroll up to those consts to understand one of the tests (I started reading them from the bottom). Plus, I needed to check whether I'm right that these consts contain 2 items.

Reply:
OTOH, I can see that in the ZWJ tests we have 5 items in such consts, so it's more justified here. So I take my above comment back. It's fine the way you coded it.